File: src/main/java/com/polykhel/billnext/service/dto/CategoryDTO.java
package com.polykhel.billnext.service.dto;

import com.polykhel.billnext.domain.enumeration.ActivityType;
import java.io.Serializable;
import java.util.Objects;
import javax.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

/**
 * A DTO for the {@link com.polykhel.billnext.domain.Category} entity.
 */
@Getter
@Setter
public class CategoryDTO implements Serializable {

    @NotNull
    private Long id;

    @NotNull
    private String name;

    private ActivityType type;

    private UserDTO user;

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof CategoryDTO)) {
            return false;
        }

        CategoryDTO categoryDTO = (CategoryDTO) o;
        if (this.id == null) {
            return false;
        }
        return Objects.equals(this.id, categoryDTO.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.id);
    }

    // prettier-ignore
    @Override
    public String toString() {
        return "CategoryDTO{" +
            "id=" + getId() +
            ", name='" + getName() + "'" +
            ", type='" + getType() + "'" +
            ", user='" + getUser() + "'" +
            "}";
    }
}

File: src/main/kotlin/com/polykhel/billnext/service/dto/CategoryDTO.kt
package com.polykhel.billnext.service.dto

import com.polykhel.billnext.domain.enumeration.ActivityType
import java.io.Serializable
import java.util.*
import javax.validation.constraints.NotNull

/**
 * A DTO for the [com.polykhel.billnext.domain.Category] entity.
 */
data class CategoryDTO(

    var id: Long? = null,

    @get: NotNull
    var name: String? = null,

    var type: ActivityType? = null,

    var user: UserDTO? = null
) : Serializable {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is CategoryDTO) return false
        val categoryDTO = other
        if (this.id == null) {
            return false
        }
        return Objects.equals(this.id, categoryDTO.id)
    }

    override fun hashCode() = Objects.hash(this.id)
}
